KSystemInformer/protection.c

```diff
@@ -668,10 +668,12 @@ VOID KphApplyObProtections(
         allowedAccessMask = process->ProcessAllowedMask;
 
         if (KphpShouldPermitCreatorProcess(Info, actor, process))
-        {
-            allowedAccessMask |= (KPH_PROCESS_READ_ACCESS |
-                                  PROCESS_TERMINATE |
-                                  PROCESS_VM_OPERATION |
+        {
+            //
+            // add permissions which are not present in the usual KPH_PROTECED_PROCESS_MASK
+            //
+
+            allowedAccessMask |= (PROCESS_VM_OPERATION |
```

Comment:
I understand your intention for the change here but it conflicts with the design. While `ProcessAllowedMask` does in fact have what was removed set by way of `KPH_PROTECTED_PROCESS_MASK`, at this stage we are using a configured variable in the process context and this code should remain indifferent to what is set there. In other words, this code doesn't need knowledge of `KPH_PROTECTED_PROCESS_MASK`, and should remain that way.

There is no streamlining or savings of instructions here as the check for the creator still occurs along with masking-in of extra bits.

This design was chosen such that the reader doesn't need to know of external information. They need only read this block of code and understand that the creator process requires these permissions to function correctly. Furthermore, it minimizes the potential for mistake in the future - consider if the permissions in `KPH_PROTECTED_PROCESS_MASK` are adjusted or changed, this code would have to be known of and potentially adjusted too. The previous code functions all the same regardless of what is set. Now, with the change, if the code elsewhere changes a subtle bug might be introduced that could be overlooked.

An "optimal" implementation would do away with the masks inside of the context and rely only on the hard coded defines for a "verified" process. However, that design workflow severely restricts the extensibility of the code. A good example of this is when suppression of these protections was added when the machine is configured in debug mode (for developers). See this commit: https://github.com/winsiderss/systeminformer/commit/8ba24f6ef590b3ee58219286792349aa7541b5e8 - you'll see in that commit that the code here inside `protection.c` needn't change at all. At that time I had full confidence that making that change would not break this workflow downstream. With this change, I can no longer have full confidence that changing code elsewhere won't break this code.